### DISCUSSION TITLE
Cannot link multiple issues to be closed in Pull Request Sidebar

### DISCUSSION THREAD

**Discussion by tcrasset** [community/community#62657] [Pull Requests]
### Select Topic Area

Bug

### Body

Hello,

Myself and some colleagues have this bug occurring on most of our pull requests.
We use the ["close-issue-on-pull-request-merge"](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) feature a lot, however it seems that linking multiple issues sometimes works, most of the time it doesn't.

If I write "Closes" followed by multiple issue numbers in the body of PR, only the first one gets listed as to-be-closed by the PR in the sidebar. So far, that's expected, as once has to repeat the "closes" keyword to list every issue .e.g `closes #1, closes #2` etc...

However, trying to then manually add the issue in the sidebar, e.g. searching by the issue number of the text, does not show the issue in the list.

Screenshots below.

I have multiple issues that start with "#147", but only the first one is shown in the list as being selected, and I can't select the ones I actually want e.g. #1476
![Screenshot 2023-08-04 at 13-10-15 feat(web) separate metrics computation from avatarization by tcrasset · Pull Request #1554 · octopize_avatar](https://github.com/community/community/assets/25140344/19710992-3a40-455c-86a3-6244ade6e0ee)
![Screenshot 2023-08-04 at 13-09-25 feat(web) separate metrics computation from avatarization by tcrasset · Pull Request #1554 · octopize_avatar](https://github.com/community/community/assets/25140344/f88b8070-3d68-4b3b-9080-accbd2820662)

**Comment by ZeeshanMukhtar1** [comment]
Hello @tcrasset ,

I understand that you and your colleagues are facing an issue with the "close-issue-on-pull-request-merge" feature on GitHub, specifically when trying to link multiple issues in the pull request body. I'd be glad to assist you in resolving this matter.

To link multiple issues in the pull request body and have them correctly displayed in the sidebar for closure upon merge, you must use the appropriate syntax. Ensure that you repeat the "closes" keyword for each issue, separated by commas. For example:

`Closes #1, Closes #2, Closes #3`
By using this format, GitHub should list each issue correctly so as to be closed by the pull request upon successful merging.

Please verify that you are using the correct issue numbers when linking them in the pull request. Make sure the issue numbers you provided match the actual issue numbers of the corresponding issues you want to link.

Additionally, confirm that the issues you are trying to link are open issues in the repository. Closed issues may not appear in the list of issues available for linking.

If you encounter any issues with manually adding issues in the sidebar, consider refreshing the page to see if it resolves the problem.

Finally, double-check the repository settings to ensure that the "close-issue-on-pull-request-merge" feature is enabled. If it's not, enable it and test again to see if the issue is resolved.

If the problem persists despite trying these steps, I recommend reaching out to GitHub support or checking their community forums for similar issues and solutions. Remember to include the steps you've taken and any error messages you encountered when seeking further assistance.

I hope this helps you to resolve the issue of linking multiple issues in your pull requests smoothly.

**Reply by Pwuts** [reply]
**This answer does not address the problem.** The reported issue is with the "Link issues" dialog, which only works partially. I'm having the same problem, and it is extremely annoying.

For example, I'm trying to link issue [Significant-Gravitas/AutoGPT#4105](https://github.com/Significant-Gravitas/AutoGPT/issues/4105) from PR [#5612](https://github.com/Significant-Gravitas/AutoGPT/pull/5612), and it just doesn't show up in the dialog:
![image](https://github.com/community/community/assets/12185583/145fc5ee-5ee1-4501-a4c0-6887592ffb87)
